src/core.php

```diff
@@ -901,7 +1004,7 @@ protected function &__xhpChildrenDeclaration(): mixed {
   abstract protected function stringify(): string;
 
   final public function toString(): string {
-    $this->__flushElementChildren();
+    $this->__flushElementChildren()->join();
```

Comment:
Should probably add a new public async function for this, and make toString() call $this->newFunc()->join()